ros/src/peripherals/src/power_board.cpp

```diff
@@ -27,222 +20,46 @@ using PowerEnableRes = peripherals::power_enable::Response;
-        ROS_ERROR("Humidity data is invalid.");
-        return false;
+        ret = false;
     }
 
-    // Populate message with water sensor data
-    valid_size = this->write("WTR", water.get(), RESPONSE_SIZE_WTR, "\r\n") == RESPONSE_SIZE_WTR;
-
-    if (valid_size)
-    {
-        eol_exists = std::memcmp(&(water[RESPONSE_SIZE_WTR-2]), "\r\n", 2) == 0;
-    }
-    if (valid_size && eol_exists)
-    {
-        msg.water_sensor = (water[1] << 8) | (water[0]);
-    }
-    else {
-        ROS_ERROR("Water data is invalid.");
-        return false;
-    }
-
-    // Populate message with main housing pressure data
-    valid_size = this->write("PIN", pressure_internal.get(), RESPONSE_SIZE_PIN, "\r\n") == RESPONSE_SIZE_PIN;
-
-    if (valid_size)
-    {
-        eol_exists = std::memcmp(&(pressure_internal[RESPONSE_SIZE_PIN-2]), "\r\n", 2) == 0;
-    }
-    if (valid_size && eol_exists)
-    {
-        msg.internal_pressure = (pressure_internal[2] << 16) | (pressure_internal[1] << 8) | (pressure_internal[0]);
-        // No conversions needed, pressure is in Pa
-    }
-    else {
-        ROS_ERROR("Internal housing pressure data is invalid.");
-        return false;
-    }
-
-    // Populate message with external water pressure data
-    valid_size = this->write("PEX", pressure_external.get(), RESPONSE_SIZE_PEX, "\r\n") == RESPONSE_SIZE_PEX;
-
-    if (valid_size)
-    {
-        eol_exists = std::memcmp(&(pressure_external[RESPONSE_SIZE_PEX-2]), "\r\n", 2) == 0;
-    }
-    if (valid_size && eol_exists)
-    {
-        msg.external_pressure = (pressure_external[1] << 8) | (pressure_external[0]);
-        // Convert from 0.01psi to Pa
-        msg.external_pressure *= PEX_TO_PASCAL_MUL;
-    }
-    else {
-        ROS_ERROR("External water pressure data is invalid.");
-        return false;
-    }
-
-    return true;
-}
-
-bool power_board::power_enabler(PowerEnableReq &req, PowerEnableRes &res)
-{
-    // Command structure: PXEb -> X is the system, b is either 0 or 1
-    // X can be: M for motors, 5 for 5V, T or 9 for the 9V/12V rail, or S for system
-    // Enable/Disable Power to Motors
-    std::string out = "PME" + std::string(req.motor_pwr_enable ? "1" : "0");
-    write(out);
-
-    // Enable/Disable 5V Rail
-    out.replace(1, 1, "5"); // Replace X position with 5 for 5V rail
-    out.replace(3, 1, req.rail_5V_pwr_enable ? "1" : "0"); // Replace b position with either 0 or 1
-    write(out);
-
-    // Enable/Disable 12V/9V Rails
-    out.replace(1, 1, "T"); // Replace X position with T for 9V/12V rail
-    out.replace(3, 1, req.rail_12V_9V_pwr_enable ? "1" : "0"); // Replace b position with either 0 or 1
-    write(out);
-
-    // Command structure: BPb -> b is either 0 or 1
-    // Enable/Disable Running Batteries in Parallel
-    out = "BP" + std::string(req.parallel_batteries_enable ? "1" : "0");
-    write(out);
-
-    return true;
+    return ret;
 }
 
 bool power_board::average_ext_pressure(AvgDataReq &req, AvgDataRes &res)
```

Comment:
We should get rid of this service and just publish the average, or have the Power Board calculate the average.

Reply:
Sounds good.